pkg/kube/client.go

```diff
@@ -884,6 +888,39 @@ func (c *client) GetIstioVersions(ctx context.Context, namespace string) (*versi
 	return &res, errs
 }
 
+func (c *client) GetProxyPods(ctx context.Context, limit int, token string) (*v1.PodList, error) {
+	params := make(map[string]string)
+	if c.revision != "" {
+		params["labelSelector"] = fmt.Sprintf("%s,%s=%s", label.ServiceCanonicalName.Name, label.IoIstioRev.Name, c.revision)
+	} else {
+		params["labelSelector"] = label.ServiceCanonicalName.Name
+	}
+	params["fieldSelector"] = "status.phase=Running"
+	// At this moment, just using a fixed limit.
+	params["limit"] = strconv.Itoa(limit)
+	if len(token) > 0 {
+		params["continue"] = token
+	}
+
+	req := c.restClient.Get().
+		Resource("pods")
+	for k, v := range params {
+		req.Param(k, v)
+	}
+
+	res := req.Do(ctx)
+	if res.Error() != nil {
+		return nil, fmt.Errorf("unable to retrieve Pods: %v", res.Error())
+	}
+
+	list := &v1.PodList{}
+	if err := res.Into(list); err != nil {
```

Comment:
why not use k8s client-go warpped api? kubernetes.Interface

Reply:
Done!